client/assets/python/aichat/utils.py

```
@@ -0,0 +1,192 @@
+    """
+    if not os.path.exists(archive_path):
+        return False
+        
+    print(f"[LOADER] Found archive at {archive_path}. Extracting...")
+    try:
+        # Create the target directory
+        os.makedirs(target_dir, exist_ok=True)
+        
+        # Extract the archive (auto-detects compression format)
+        with tarfile.open(archive_path, 'r:*') as tar:
+            tar.extractall(path=target_dir)
```

Ruff (0.14.5)

Uses of `tarfile.extractall()`

(S202)